provisioning/device/test/_transport_state_machine_test.js

```diff
@@ -0,0 +1,502 @@
+var registrationRequestReturnsFailure = function () { return sinon.spy(function (registrationId, auth, body, force, callback) { callback(new Error(fakeErrorText)); }); };
+var registrationRequestReturnsBadResponse = function () { return sinon.spy(function (registrationId, auth, body, force, callback) { callback(null, { status: fakeBadStatus }); }); };
+var operationStatusReturnsAssigned = function () { return sinon.spy(function (registrationId, operationId, callback) { callback(null, { status: 'Assigned' } ); }); };
+var operationStatusReturnsFailure = function () { return sinon.spy(function (registrationId, operationId, callback) { callback(new Error(fakeErrorText)); }); };
+var operationStatusReturnsBadResponse = function () { return sinon.spy(function (registrationId, operationId, callback) { callback(null, { status: fakeBadStatus }); }); };
+var operationStatusReturnsAssigningThenAssigned = function () {
+  var callCount = 0;
+  return sinon.spy(function (registrationId, operationId, callback) {
+    callCount++;
+    if (callCount === 1) {
+      callback(null, { status: 'Assigning' }, null, 0);
+    } else if (callCount === 2) {
+      callback(null, { status: 'Assigned' });
+    } else {
+      assert(false, 'second status query returned Assigned.  There should not be a third status query');
+    }
+  });
+};
+
+describe('state machine', function () {
+  this.timeout(1000);
+
+  var makeNewMachine = function () {
+    /* Tests_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_001: [ The `constructor` shall accept no arguments ] */
+    var machine = new TransportStateMachine();
+    machine.initialize({
+      connect: sinon.stub().callsArg(0),
+      disconnect: sinon.stub().callsArg(0),
+      registrationRequest: registrationRequestReturnsAssigned(),
+      queryOperationStatus: operationStatusReturnsAssigned(),
+      getErrorResult:sinon.stub().callsArg(0)
+    });
+    return machine;
+  };
+
+  var clearTransportCallCount = function (machine) {
+    machine._transport.connect.reset();
+    machine._transport.disconnect.reset();
+    machine._transport.registrationRequest.reset();
+    machine._transport.queryOperationStatus.reset();
+    machine._transport.getErrorResult.reset();
+  };
+
+  var assertNoTransportFunctionsCalled = function (machine) {
+    assert.isFalse(machine._transport.connect.called);
+    assert.isFalse(machine._transport.disconnect.called);
+    assert.isFalse(machine._transport.registrationRequest.called);
+    assert.isFalse(machine._transport.queryOperationStatus.called);
+    assert.isFalse(machine._transport.getErrorResult.called);
+  };
+
+  var callRegisterWithDefaultArgs = function (callback) {
+    machine.register(fakeRegistrationId, fakeAuth, fakeRequestBody, false, callback);
+  };
+
+  var machine;
+  beforeEach(function () {
+    machine = makeNewMachine();
```

Comment:
>machine [](start = 4, length = 7)

here you could have a transport object that you pass to makeNewMachine.
then you'd have a public transport object on which you can assert stuff without reaching into machine._transport.